crypto/modes/gcm128.c

```diff
@@ -537,7 +537,7 @@ static void gcm_ghash_4bit(u64 Xi[2], const u128 Htable[16],
             Xi[0] = Z.hi;
             Xi[1] = Z.lo;
         }
-    } while (inp += 16, len -= 16);
+    } while (inp += 16, (len >= 16) && (len -= 16));
```

Comment:
Can't say that I'm a fan of this expression.  The best I can come up with is:
```c
        inp += 16;
        len -= 16;
    } while (len > 0);
```
which isn't a great stopping criteria (but just as good as the original).